compiler/extensions/cpp/runtime/src/zserio/BitStreamReader.cpp

```diff
@@ -229,6 +236,8 @@ namespace
                 cacheBuffer = parse8(ctx.buffer + byteIndex);
                 break;
             }
+
+            cacheBuffer >>= alignedBitSize - ctx.cacheNumBits;
```

Comment:
@Mi-La I've changed `ctx.cacheNumBits` back to the unaligned value.
By r-shifting the cache buffer by the difference between aligned and unaligned the unneeded/invalid bits get shifted out.